### DISCUSSION TITLE
Are there any news about Spyglass?

### DISCUSSION THREAD

**Discussion by makp0** [Eventuous/eventuous#233] [General]
Wondering what's the state of the Spyglass project. Will it be open-sourced? Are there any timeline?

**Comment by alexeyzimarev** [comment]
It won't be open-sourced, at least I don't plan to do so. It's a bit stale now, but it's usable in a way I demoed it. I plan some refactoring there so it can be more serverless-friendly.

**Comment by makp0** [comment]
Thank you for your response. 
Is it available anywhere to use it as a service or executable?
Are you planning to license it?

**Reply by alexeyzimarev** [reply]
It's better to reply to my answer, now your follow up question looks like an answer :)
You can use it at http://spyglass.eventuous.cloud, there's no fee, but it requires registration. I have no immediate plans to make any money out of it, and what's available there _now_ will always be free to use.

**Reply by alexeyzimarev** [reply]
You'd need a tunnel if you are inspecting an app that is not available via public internet, ngrok is a good option. My refactoring plan includes running it fully in the browser, so it should be playing better with local/private apps.

**Reply by makp0** [reply]
Sorry, didn't pay much attention while replying on the phone. For sure, making threads > piling up replies.

Unbelievable, I turned whole internet upside down searching Spyglass mentions, code or anything, found none. How long was it deployed?

Btw, after doing some research on tunnels, my findings:
N1 exposing tunnel -  https://expose.dev/. as opposed to ngrok, does not limit session duration on free tier.
(not related: N1 private tunnel - https://www.zerotier.com/)

**Reply by alexeyzimarev** [reply]
Oh, cool, I didn't know about any of those tools. Expose definitely looks like a good alternative, and ZeroTier sounds more like Tailscale if I get it right.

I thought about embedding the tunnelling feature to the NuGet package (it's open source, part of the repository), and ngrok started to release embeddable agent, but the .NET version isn't there (they told me it's coming). Something like ZeroTier could be an option, but it requires complex deployment as the last thing one would want is to share their network with strangers :) It would be more "enterprise", and I don't think I am ready for such challenge yet :)

As for "where Spyglass runs", I mentioned the URL during my demo (video in the Eventuous YouTube channel) and during my talk at KanDDDinsky last year. I deliberately didn't want to advertise it in public because it uses Server-side Blazor and runs in Cloud Run, so if someone leaves the browser open, the app will be running and I'd have to pay the fee. That's why I plan changing the internals so I won't face the risk of unexpected costs. Pure web assembly could work, another alternative is to pack what's already built in a desktop app. I tried the last option, but it's a bit weird, doesn't look exactly as in the browser. Maybe I can change some settings.

**Comment by gius** [comment]
Hi, do you have any update on this topic? It would be great to be able to run Spyglass locally!

**Reply by alexeyzimarev** [reply]
I need to wrap up the docs migration, and the docs for 0.15 before I can do anything else. Sorry. The plan is to convert it to a browser-WASM, so it will work.
